### PR TITLE
docs: revamp Sign in with Apple and Google docs

### PR DESCRIPTION
**Pull request by hf**
The new docs explain how to use all the ways Apple can be used:
- On the web
- With iOS, macOS, watchOS, tvOS apps
- With Sign in with Apple JS

Similarly for Google:
- On the web, including with One Tap
- With Android apps

### CODE REVIEW

**Review by gregnr — APPROVED** [review]
Nice work!

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 1]
```diff
@@ -3,119 +3,215 @@ import Layout from '~/layouts/DefaultGuideLayout'
...
+3. Finally, add the client ID and secret from step 1 in the [Google provider on the Supabase Dashboard](https://supabase.com/dashboard/project/_/auth/providers). 
 
-Copy your new OAuth credentials
+## Using native sign in
 
-- A box will appear called `OAuth client created`.
-- Copy and save the values under `Your Client ID` and `Your Client Secret`.
+Unlike the OAuth flow which requires the use of a web browser, the native Sign in with Google flow on Android uses the [operating system's built-in functionalities](https://developers.google.com/android/reference/com/google/android/gms/auth/api/identity/package-summary) to prompt the user for consent. Note that native sign-in has been rebranded as _One Tap sign-in on Android_ by Google, which you should not confuse with _One Tap sign in for web_, as mentioned below.
```
Comment (dshukertjr) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:68]: Do we want to mention native Google sign in for iOS as well?
https://developers.google.com/identity/sign-in/ios/start

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 2]
```diff
@@ -3,119 +3,215 @@ import Layout from '~/layouts/DefaultGuideLayout'
...
 
-When your user signs out, call [signOut()](/docs/reference/javascript/auth-signout) to remove them from the browser session and any objects from localStorage:
+For apps written in Kotlin, please consult the [community maintained library](/docs/reference/kotlin/introduction).
 
-```js
-async function signout() {
-  const { error } = await supabase.auth.signOut()
+### Configuration
+
+1. Configure OAuth credentials for your Google Cloud project in the [Credentials](https://console.cloud.google.com/apis/credentials) page of the console. When creating a new OAuth client ID, choose _Android_ for the application type. Use the instructions on screen to provide the SHA-1 certificate fingerprint used to sign your Android app.
```
Comment (dshukertjr) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:96]: Here as well, we could probably rephrase it to something like "choose which ever platform you want to add Google sign in to, Android or iOS for the application type." to acknowledge the iOS Google login.

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 3]
```diff
@@ -3,119 +3,215 @@ import Layout from '~/layouts/DefaultGuideLayout'
...
 
-## Enter your Google credentials into your Supabase Project
+When the user provides consent, Google issues an identity token (commonly abbreviated as ID token) that is then sent to your project's Supabase Auth server. When valid, a new user session is started by issuing an access and refresh token from Supabase Auth.
 
-<SocialProviderSettingsSupabase provider="Google" />
+If you're building an Android app using Flutter, you can use `signInWithGoogle()` to quickly get the user's consent and sign them in to your Supabase project:
 
-## Add login code to your client app
+```dart
+final AuthResponse = await supabase.auth.signInWithGoogle();
```
Comment (dshukertjr) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:75]: Just an minor update, but we decided that the iOS Client ID and Android Client ID will be passed as parameters like this.
```suggestion
final AuthResponse = await supabase.auth.signInWithGoogle(
  iosClientId: 'IOS_CLIENT_ID',
  androidClientId: 'ANDROID_CLIENT_ID',
);
```

**Review by dshukertjr — COMMENTED** [review]
@hf I have updated the code for Flutter.

**Review by dshukertjr — APPROVED** [review]

File: apps/docs/pages/guides/auth/social-login/auth-google.mdx [hunk 4]
```diff
@@ -3,119 +3,217 @@ import Layout from '~/layouts/DefaultGuideLayout'
...
 
-- A box will appear called `OAuth client created`.
-- Copy and save the values under `Your Client ID` and `Your Client Secret`.
+Unlike the OAuth flow which requires the use of a web browser, the native Sign in with Google flow on Android uses the [operating system's built-in functionalities](https://developers.google.com/android/reference/com/google/android/gms/auth/api/identity/package-summary) to prompt the user for consent. Note that native sign-in has been rebranded as _One Tap sign-in on Android_ by Google, which you should not confuse with _One Tap sign in for web_, as mentioned below.
 
-## Enter your Google credentials into your Supabase Project
+When the user provides consent, Google issues an identity token (commonly abbreviated as ID token) that is then sent to your project's Supabase Auth server. When valid, a new user session is started by issuing an access and refresh token from Supabase Auth.
 
-<SocialProviderSettingsSupabase provider="Google" />
+If you are building a Flutter app, you can use [flutter_appauth](https://pub.dev/packages/flutter_appauth) package to sign a user into your Supabase project:
```
Comment (dshukertjr) [inline apps/docs/pages/guides/auth/social-login/auth-google.mdx:72]:
```suggestion
```
